src/main/java/xyz/nucleoid/creator_tools/item/RegionVisibilityFilterItem.java

```diff
@@ -0,0 +1,87 @@
+package xyz.nucleoid.creator_tools.item;
+
+import java.util.List;
+
+import eu.pb4.polymer.api.item.PolymerItem;
+import net.minecraft.client.item.TooltipContext;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.nbt.NbtElement;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.text.Text;
+import net.minecraft.util.Formatting;
+import net.minecraft.util.Hand;
+import net.minecraft.util.TypedActionResult;
+import net.minecraft.world.World;
+import xyz.nucleoid.creator_tools.workspace.MapWorkspaceManager;
+import xyz.nucleoid.creator_tools.workspace.editor.ServersideWorkspaceEditor;
+
+public final class RegionVisibilityFilterItem extends Item implements PolymerItem {
+    private static final String REGION_KEY = "Region";
+
+    public RegionVisibilityFilterItem(Settings settings) {
+        super(settings);
+    }
+
+    @Override
+    public TypedActionResult<ItemStack> use(World world, PlayerEntity player, Hand hand) {
+        if (world.isClient()) {
+            return super.use(world, player, hand);
+        }
+
+        var stack = player.getStackInHand(hand);
+
+        if (player instanceof ServerPlayerEntity serverPlayer) {
+            var workspaceManager = MapWorkspaceManager.get(serverPlayer.server);
+            var editor = workspaceManager.getEditorFor(serverPlayer);
+
+            var regions = getRegions(stack);
+
+            if (editor != null && editor.useRegionVisibilityFilterItem(regions)) {
+                return TypedActionResult.success(stack);
+            }
+        }
+
+        return TypedActionResult.pass(stack);
+    }
+
+    @Override
+    public Item getPolymerItem(ItemStack stack, ServerPlayerEntity player) {
+        return Items.LEATHER_LEGGINGS;
+    }
+
+    @Override
+    public int getPolymerArmorColor(ItemStack stack, ServerPlayerEntity player) {
+        var regions = getRegions(stack);
+        if (regions == null || regions.isEmpty()) return -1;
+
+        var region = regions.get(0);
+        return ServersideWorkspaceEditor.colorForRegionMarker(region);
+    }
+
+    @Override
+    public void appendTooltip(ItemStack stack, World world, List<Text> tooltip, TooltipContext context) {
+        var regions = getRegions(stack);
+
+        if (regions != null) {
+            for (var region : regions) {
+                tooltip.add(Text.literal(region).formatted(Formatting.GRAY));
+            }
+        }
+    }
+
+    private static List<String> getRegions(ItemStack stack) {
+        var nbt = stack.getNbt();
+        if (nbt == null) return null;
```

Comment:
Nit: maybe just returning `List.of()` instead of these null cases is reasonable here? Otherwise, `@Nullable`?

Reply:
I used `null` to differentiate from explicitly filtering all regions with `{Region: []}`.

Reply:
I added the `@Nullable` annotation in 1fb476ab0232e78158359a7b7fcf9076d4a1c4db.